seqr/views/utils/dataset_utils.py

```diff
@@ -94,37 +94,45 @@ def load_mapping_file_content(file_content):
         id_mapping[line[0]] = line[1]
     return id_mapping
 
+
+def _get_individual_sample_lookup(individuals):
+    return {i.individual_id: i for i in individuals}
+
+
+def _get_mapped_individual_lookup_key(sample_id_to_individual_id_mapping):
+    def _get_mapped_id(sample_id):
+        return sample_id_to_individual_id_mapping.get(sample_id, sample_id)
+    return _get_mapped_id
+
+
 def _find_or_create_missing_sample_records(
         samples,
         projects,
         user,
         sample_ids,
-        sample_id_to_individual_id_mapping=None,
         raise_no_match_error=False,
         raise_unmatched_error_template=None,
         create_active=False,
+        get_individual_sample_lookup=_get_individual_sample_lookup,
```

Comment:
I see what you mean. It makes it ready for adding project name filters for RNAseq data.